### PR TITLE
Chore/1653 full asset details

### LINKED ISSUE #1653: Show full asset details in market info and dialog
## The Chore

There is missing asset data that we should show in the market info tab and the asset dialog

#### Market info
- [x] decimals
- [x] quantum
- [x] type (IE typename, Builtin or ERC20)
- [x] contractAddress (for ERC20)
- [x] lifetimeLimit with tooltip (for ERC20)
- [x] withdrawThreshold with tooltip (for ERC20)
- [x] maxFaucetAmountMint  (for Builtin)
- [x] status
- [x] infrastructureFeeAccount balance with tooltip
- [x] globalRewardPoolAccount balance with tooltip
- [x] takerFeeAccount balance with tooltip
- [x] makerFeeRewardAccount balance with tooltip
- [x] lpFeeRewardAccount balance with tooltip
- [x] marketProposerRewardAccoiunt balance with tooltip

#### Asset dialog
- [x] ID
- [x] type (IE typename, Builtin or ERC20)
- [x] status
- [x] infrastructureFeeAccount balance with tooltip
- [x] globalRewardPoolAccount balance with tooltip
- [x] takerFeeAccount balance with tooltip
- [x] makerFeeRewardAccount balance with tooltip
- [x] lpFeeRewardAccount balance with tooltip
- [x] marketProposerRewardAccoiunt balance with tooltip

### PR DESCRIPTION
**Pull request by asiaznik**
# Related issues 🔗

Closes #1653 

# Description ℹ️

Details dialog and market info tab show not the full set of asset details.

# Demo 📺

Screenshot 2022-10-10 at 20 05 14

Screenshot 2022-10-10 at 20 05 05

### CODE REVIEW

File: libs/assets/src/lib/asset-details-table.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        { key: AssetDetail.LIFETIME_LIMIT, value: '123,000.000' },
+        {
+          key: AssetDetail.INFRASTRUCTURE_FEE_ACCOUNT_BALANCE,
+          value: '0.001',
+        },
+        {
+          key: AssetDetail.GLOBAL_REWARD_POOL_ACCOUNT_BALANCE,
+          value: '0.002',
+        },
+        { key: AssetDetail.TAKER_FEE_REWARD_ACCOUNT_BALANCE, value: '0.003' },
```
Comment (candida-d) [inline libs/assets/src/lib/asset-details-table.spec.tsx:36]: 'Taker fee reward' and 'maker fee reward' have been renamed

File: libs/assets/src/lib/asset-details-table.tsx [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+  // erc20 details:
+  CONTRACT_ADDRESS,
+  WITHDRAWAL_THRESHOLD,
+  LIFETIME_LIMIT,
+  // builtin details:
+  MAX_FAUCET_AMOUNT_MINT,
+  // balances:
+  INFRASTRUCTURE_FEE_ACCOUNT_BALANCE,
+  GLOBAL_REWARD_POOL_ACCOUNT_BALANCE,
+  TAKER_FEE_REWARD_ACCOUNT_BALANCE,
```
Comment (candida-d) [inline libs/assets/src/lib/asset-details-table.tsx:36]: no more 'taker/maker fee reward' https://docs.vega.xyz/docs/testnet/graphql/enums/account-type

File: libs/assets/src/lib/asset-details-table.tsx [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+  {
+    key: AssetDetail.DECIMALS,
+    label: t('Decimals'),
+    tooltip: t('Number of decimal / precision handled by this asset'),
+    value: (asset) => asset.decimals.toString(),
+  },
+  {
+    key: AssetDetail.QUANTUM,
+    label: t('Quantum'),
+    tooltip: t('The minimum economically meaningful amount in the asset'),
```
Comment (candida-d) [inline libs/assets/src/lib/asset-details-table.tsx:84]:
```suggestion
    tooltip: t('The minimum economically meaningful amount of the asset'),
```

File: libs/assets/src/lib/asset-details-table.tsx [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    tooltip: t(
+      'The address of the contract for the token, on the ethereum network'
+    ),
+    value: (asset) => (asset.source as Schema.ERC20).contractAddress,
+  },
+  {
+    key: AssetDetail.WITHDRAWAL_THRESHOLD,
+    label: t('Withdrawal threshold'),
+    tooltip: t(
+      'The maximum allowed per withdraw note: this is a temporary measure for restricted mainnet'
```
Comment (candida-d) [inline libs/assets/src/lib/asset-details-table.tsx:106]:
```suggestion
      'The maximum allowed per withdrawal. Note: this is a temporary measure for restricted mainnet'
```

File: libs/assets/src/lib/asset-details-table.tsx [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+    ),
+    value: (asset) =>
+      num(asset, (asset.source as Schema.ERC20).withdrawThreshold),
+  },
+  {
+    key: AssetDetail.LIFETIME_LIMIT,
+    label: t('Lifetime limit'),
+    tooltip: t(
+      'The lifetime limits deposit per address note: this is a temporary measure for restricted mainnet'
```
Comment (candida-d) [inline libs/assets/src/lib/asset-details-table.tsx:115]:
```suggestion
      'The lifetime deposit limit per address. Note: this is a temporary measure for restricted mainnet'
```

File: libs/assets/src/lib/asset-details-table.tsx [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+    value: (asset) => num(asset, asset.infrastructureFeeAccount.balance),
+  },
+  {
+    key: AssetDetail.GLOBAL_REWARD_POOL_ACCOUNT_BALANCE,
+    label: t('Global reward pool account balance'),
+    tooltip: t('The global reward pool account for this asset'),
+    value: (asset) => num(asset, asset.globalRewardPoolAccount?.balance),
+  },
+  {
+    key: AssetDetail.TAKER_FEE_REWARD_ACCOUNT_BALANCE,
```
Comment (candida-d) [inline libs/assets/src/lib/asset-details-table.tsx:141]: no more taker fee reward

File: libs/assets/src/lib/asset-details-table.tsx [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+    label: t('Taker fee reward account balance'),
+    tooltip: t('The taker fee reward account for this asset'),
+    value: (asset) => num(asset, asset.takerFeeRewardAccount?.balance),
+  },
+  {
+    key: AssetDetail.MAKER_FEE_REWARD_ACCOUNT_BALANCE,
```
Comment (candida-d) [inline libs/assets/src/lib/asset-details-table.tsx:147]: update to either maker fees paid or received

**Review by MadalinaRaicu — APPROVED** [review]

File: libs/assets/src/lib/asset-details-table.tsx [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+    label: t('Maker fee reward account balance'),
+    tooltip: t('The maker fee reward account for this asset'),
+    value: (asset) => num(asset, asset.makerFeeRewardAccount?.balance),
+  },
+  {
+    key: AssetDetail.LP_FEE_REWARD_ACCOUNT_BALANCE,
+    label: t('Liquidity provision fee reward account balance'),
+    tooltip: t('The liquidity provision reward account for this asset'),
```
Comment (candida-d) [inline libs/assets/src/lib/asset-details-table.tsx:155]: This tooltip (and some others in this section) can work a little harder. Maybe something like "The rewards acquired based on the liquidity provision fees you've received".

File: libs/assets/src/lib/asset-details-table.tsx [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+];
+
+const AssetStatusMapping: Mapping = {
+  STATUS_ENABLED: {
+    value: t('Enabled'),
+    tooltip: t('Asset can be used on the Vega network'),
+  },
+  STATUS_PENDING_LISTING: {
+    value: t('Pending listing'),
+    tooltip: t('Asset is pending listing on the ethereum bridge'),
```
Comment (candida-d) [inline libs/assets/src/lib/asset-details-table.tsx:173]:
```suggestion
    tooltip: t('Asset needs to be added to the Ethereum bridge'),
```

File: libs/assets/src/lib/asset-details-table.tsx [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+  },
+  STATUS_PROPOSED: {
+    value: t('Proposed'),
+    tooltip: t('Asset is proposed to be added to the network'),
```
Comment (candida-d) [inline libs/assets/src/lib/asset-details-table.tsx:177]:
```suggestion
    tooltip: t('Asset has been proposed to the network'),
```

File: libs/assets/src/lib/Asset.graphql [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    }
+  }
+  status
+  infrastructureFeeAccount {
+    balance
+  }
+  globalRewardPoolAccount {
+    balance
+  }
+  takerFeeRewardAccount {
```
[the remark this answers is not in the document]
Comment (asiaznik) [reply libs/assets/src/lib/Asset.graphql:25]: It was left in the API as `takerFeeRewardAccount` but gets the `AccountType_ACCOUNT_TYPE_REWARD_MAKER_PAID_FEES` account.

```
func (r *myAssetResolver) TakerFeeRewardAccount(ctx context.Context, asset *types.Asset) (*types.Account, error) {
	return listAssetAccounts(ctx, r.tradingDataClientV2, asset, types.AccountType_ACCOUNT_TYPE_REWARD_MAKER_PAID_FEES)
}
```

**Pull request merged**